File: Python3/SetIntersection.py
import sys
from functools import reduce

with open(sys.argv[1], 'r') as test_cases:
    for test in test_cases:
        result = list(
            reduce(
                set.intersection,
                map(
                    lambda s: set(
                        map(int, s.split(','))
                    ),
                    test.split(';')
                )
            )
        )
        print(
            ','.join(
                map(str, sorted(result))
            )
        )

File: Elixir/SetIntersection.ex
defmodule SetIntersection do
  def setIntersection(s) when is_binary(s) do
    String.split(s, ";", trim: true)
    |> Enum.map(&(String.split &1, ",", trim: true))
    |> Enum.map(fn xs -> (Enum.map xs, &(String.to_integer &1)) end)
    |> Enum.map(&(MapSet.new &1))
    |> Enum.reduce(&(MapSet.intersection &1, &2))
    |> Enum.sort
    |> Enum.join(",")
  end
end
